### PR TITLE
SAMBRO:remove blank lines from HTML + add the text back to the label + CAP: update, cancel and clear msg type

### PR DESCRIPTION
**Pull request by biplovbhandari**
@flavour Please merge

### CODE REVIEW

File: controllers/cap.py [hunk 1]
```diff
@@ -155,6 +155,31 @@ def prep(r):
...
                              args = request.args,
                              vars = request.vars))
+            if r.record.approved_by is not None:
+                # Once approved, don't allow to edit
+                # Don't allow to delete
+                s3db.configure(tablename,
+                               editable=False,
+                               deletable=False,
+                               )
+            # Uncomment or add to this according to country CAP profile
```
Comment (flavour) [inline controllers/cap.py]: Won't hold up merge, but 'uncomment' won't be an easy option...either this needs to be in a template fragment (once we have cascading templates) or else a deployment_setting

File: modules/s3db/cap.py [hunk 2]
```diff
@@ -1842,6 +1865,11 @@ def cap_rheader(r):
                     if action_btn:
                         rheader.insert(1, TR(TD(action_btn)))
 
+                    if msg_type_btn:
```
Comment (flavour) [inline modules/s3db/cap.py]: I won't hold up merge, but I don 't see any need for this extra var?
Why not just:
if msg_error_btn:

File: modules/templates/SAMBRO/views/index.html [hunk 3]
```diff
@@ -2,11 +2,8 @@
 <div id='home'>
 <h1>{{=title}}</h1>
 <div class='subscribe-button'>
- {{if current.auth.s3_logged_in():}}
-  <a href='/{{=appname}}/cap/alert.rss' target='_blank'><i class='icon-rss'></i></a>
- {{else:}}
-  <a href='/{{=appname}}/cap/public.rss' target='_blank'><i class='icon-rss'></i></a>
- {{pass}}
+{{fn = "alert.rss" if current.auth.s3_logged_in() else "public.rss"}}
```
Comment (flavour) [inline modules/templates/SAMBRO/views/index.html]: Slightly DRYer if the .rss is not part of the fn variable (also matches the meaning of the fn better!)

**Comment by biplovbhandari** [comment]
Thanks @flavour will work on the suggestions :)

File: modules/s3db/cap.py (hunk 2 above)
Comment (nursix) [inline modules/s3db/cap.py]: I said that using any of the buttons msg_*_btn can lead to regression if you remove one of them.

But I also said that the condition itself can be assigned to a boolean variable, not to create a separate one.

**Pull request merged**